lock-api-objects/src/main/java/com/palantir/lock/v2/TimestampLeaseResult.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import java.util.function.Supplier;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface TimestampLeaseResult {
+    @Value.Parameter
+    long minLeasedTimestamp();
+
+    @Value.Parameter
+    LockToken lock();
+
+    @Value.Parameter
+    Supplier<Long> freshTimestampsSupplier();
```

Comment:
This should be LongSupplier to unbox all the things. I missed the refactor on the way. I will either modify here or add the change in a later PR in the stack.

Reply:
Yeah, was going to say. And we should propagate all the way